sktime/transformers/pca.py

```diff
@@ -0,0 +1,95 @@
+import pandas as pd
+
+from sklearn.utils.validation import check_is_fitted
+from sklearn.decomposition import PCA
+
+from sktime.transformers.base import BaseTransformer
+from sktime.utils.validation.supervised import validate_X, check_X_is_univariate
+from sktime.utils.data_container import tabularise, detabularise, check_equal_index
+
+
+class PCATransformer(BaseTransformer):
+    """ Transformer that applies Principle Components Analysis to a univariate time series.
+
+    Provides a simple wrapper around ``sklearn.decomposition.PCA``.
+
+    Parameters
+    ----------
+    n_components : int, float, str or None (default None)
+        Number of principle components to retain. By default, all components are retained. See
+        ``sklearn.decomposition.PCA`` documentation for a detailed description of all options.
+    **kwargs
+        Additional parameters passed on to ``sklearn.decomposition.PCA``. See ``sklearn.decomposition.PCA``
+        documentation for a detailed description of all options.
+    """
+
+    def __init__(self, n_components=None, **kwargs):
+        self.pca = PCA(n_components, **kwargs)
+        self.input_dim_ = None
+
+    def fit(self, X, y=None):
+        """
+        Fit transformer, finding all principal components.
+
+        Parameters
+        ----------
+        X : nested pandas DataFrame of shape [n_samples, 1]
+            Nested dataframe with univariate time-series in cells.
+
+        Returns
+        -------
+        self : an instance of self.
+        """
+
+        validate_X(X)
+        check_X_is_univariate(X)
+        check_equal_index(X)
```

Comment:
```suggestion
```
I'd remove this line, this is something the data container should check during construction, but not each time we want to transform the data. But the data container functionality has not been implemented yet, here's the discussion if you're interested: #15